packages/pluggableWidgets/bar-chart-native/src/ui/Styles.ts

```diff
@@ -0,0 +1,114 @@
+import { TextStyle, ViewStyle } from "react-native";
+
+export interface BarChartGridStyle {
+    backgroundColor?: string;
+    dashArray?: string;
+    lineColor?: string;
+    padding?: number;
+    paddingBottom?: number;
+    paddingHorizontal?: number;
+    paddingLeft?: number;
+    paddingRight?: number;
+    paddingTop?: number;
+    paddingVertical?: number;
+    width?: number;
+}
+
+export interface BarChartAxisStyle<T extends "X" | "Y"> {
+    color?: string;
+    dashArray?: string;
+    fontFamily?: string;
+    fontSize?: number;
+    fontStyle?: "normal" | "italic";
+    fontWeight?: "normal" | "bold" | "100" | "200" | "300" | "400" | "500" | "600" | "700" | "800" | "900";
+    label?: TextStyle & {
+        relativePositionGrid?: T extends "X" ? "bottom" | "right" : "top" | "left";
+    };
+    lineColor?: string;
+    width?: number;
+}
+
+interface BarChartBarStyle {
+    ending?: number;
+    barColor?: string;
+    width?: number;
+}
+
+interface BarChartBarLabelStyle {
+    // color is the same as bar color
+    fontFamily?: string;
+    fontSize?: number;
+    fontStyle?: "normal" | "italic";
+    fontWeight?: "normal" | "bold" | "100" | "200" | "300" | "400" | "500" | "600" | "700" | "800" | "900";
+}
+
+export interface BarChartLegendStyle {
+    container?: ViewStyle;
+    item?: ViewStyle;
+    indicator?: ViewStyle;
+    label?: TextStyle;
+}
+
+export interface BarChartStyle {
+    container?: ViewStyle;
+    errorMessage?: TextStyle;
+    chart?: ViewStyle;
+    grid?: BarChartGridStyle;
+    xAxis?: BarChartAxisStyle<"X">;
+    yAxis?: BarChartAxisStyle<"Y">;
+    legend?: BarChartLegendStyle;
+    domain?: {
```

Comment:
I think we should include domain styling into `xAxis` and `yAxis` styles. The reason is that on a root style object level we try to refer to widget elements (see native styling docs).

Reply:
I agree unfortunately with me leaving on holiday shortly I can't implement this ideally using `StyleUtils` mapper functions with generics. I added a [JIRA](https://mendix.atlassian.net/browse/WT-3033) for this.